### ISSUE TITLE
Referer not correct for embedded image

### ISSUE THREAD

**Issue by leonard84** [Robbendebiene/Gesturefy#434]
The fix for #393 broke the image download for sites that verify the exact referer url. I just got around to trying to figure out what causes this. I checked the code and the extension just uses the origin instead of the url, is there any reason for this? When I modified it to use `returnValues[0].url` it worked again. If there was a reason to use the origin instead of the url, I'd suggest making it configurable either globally or by site.

```
      // if the image is embedded in a website use the origin of that website as the referrer
      if (data.target.src !== returnValues[0].url) {
        queryOptions.headers = [ { name: "Referer", value: returnValues[0].origin } ];
      }
```

### Your System
- Operating system: Win10
- Firefox version: 71.0
- Gesturefy version: 2.1.3

**Comment by Robbendebiene** [comment]
> The fix for #393 broke the image download for sites that verify the exact referer url.

Are you sure that it worked before the update? Because previously I couldn't even send any referrer.

> I checked the code and the extension just uses the origin instead of the url, is there any reason for this?

The main reason is data minimisation / privacy protection, since I currently do not check the security levels. But I should better simulate the default behaviour: 

> The origin, path, and querystring of the URL are sent as a referrer when the protocol security level stays the same (HTTP→HTTP, HTTPS→HTTPS) or improves (HTTP→HTTPS), but isn't sent to less secure destinations (HTTPS→HTTP).

So by default the Referrer contains the whole URL except of the hash (`uri = window.location.href.split("#")[0];`). I'll add another check for the protocols and then send the full URI.

**Comment by Robbendebiene** [comment]
Fixed in Gesturefy version [2.1.4](https://github.com/Robbendebiene/Gesturefy/releases/tag/v2.1.4)
